python/res/fm/shell/shell.py

```
@@ -212,3 +212,9 @@ def copyFile(src , target = None):
                 shutil.copyfile( src , target_file )
         else:
             raise IOError("Input argument:'%s' does not correspond to an existing file" % src)
+
+    @staticmethod
+    def carefulCopyFile(src , target = None):
+        if os.path.exists(target):
+            raise IOError("Input argument:'%s' already exist" % target)
```

That the target file exists is perfectly OK - not exception.

---

Should the function fail silently? Would it not be a good idea to let the user know if the copy failed?

---

Well - if the file is already present the copy should not proceed, and that is the expected operation of this function - i.e. *not* a failure. Could add a message like:
```
print("File: {} already present - not updated".format(target_file))
```

---

Very good, I will change it accordingly.